### PR TITLE
logging: cleanup low level logger registration api - v5

### PR DESCRIPTION
**Pull request by jasonish**
Continues some old work from year seen here:
https://github.com/OISF/suricata/pull/9493

We've already seen on library usage hooking into the low level logging APIs
which are already reasonably designed for plugin and library users, so clean
them up a little add add some documentation.

- don't use OutputCtx in the low level loggers, this is from a higher level
  abstraction. Use void *initdata instead.
- Remove ThreadExitPrintStats from logging modules, it only had one valid use
  in a now deprecated output (tlslog).
- Add doxygen comments to the registration function in the header.
- SCPrefix the registration functions.
- Update the output section of extending Suricata in the devguide to highlight
  these functions. At this time, extra documentation over and above whats in
  doxygen is not added.
- Add a plugin to demonstrate the use of the low level logging API (for a
  couple of the loggers).

Ticket: https://redmine.openinfosecfoundation.org/issues/7227

### CODE REVIEW

File: src/suricata-common.h [hunk 1]
```diff
@@ -493,6 +493,10 @@ typedef enum {
     LOGGER_JSON_STREAM,
     LOGGER_JSON_ARP,
 
+    /* An ID that can be used by loggers registered by plugins and/or
+     * library users. */
+    LOGGER_USER,
```
Comment (jasonish) [inline src/suricata-common.h:498]: Would multiple loggers using the same ID be bad for transaction loggers, I think they are the only loggers using the logger_id for some state management. @catenacyber Have you looked at making these dynamic in any of your work?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22300

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #11689, thanks!

File: src/suricata-common.h (hunk 1 above)
Comment (catenacyber) [reply src/suricata-common.h:498]: I did not touch these IDs...

**Pull request closed without merging**